PTZ + Camera Settings

Keroi opened the pull request.

Keroi committed:
Update README.md

Keroi committed:
[Version] 0.2.1 :
- Added PTZ support
- Added Camera Settings via ipc_cmd

Keroi committed:
Merge branch 'master' of https://github.com/Keroi/yi-hack-v4

Keroi committed:
Update README.md

Keroi committed:
Fix preset.sh file missing

Keroi committed:
Merge branch 'master' of https://github.com/Keroi/yi-hack-v4

TheCrypt0 commented:
Hi @Keroi,
Thank you for the contribution!
Would you mind writing a quick sum up of what you updated in the original files and the new features?

Keroi committed:
Cleanup

Keroi commented:
Hi @TheCrypt0 !

Sure. Most of the new files are based on roleoroleo works for the MStar platform : https://github.com/roleoroleo/yi-hack-MStar/. Those changes add the ipc_cmd used for PTZ, presets and some camera settings such as rotate, IR, led, ... (copied from official app) :
- src/ipc_cmd/*
- src/static/static/home/yi-hack-v4/etc/camera.conf (those are the default values for the camera settings)

All that stuff is for the new pages (PTZ and Camera Settings) :
- src/www/httpd/cgi-bin/camera_settings.sh
- src/www/httpd/cgi-bin/preset.sh 
- src/www/httpd/cgi-bin/ptz.sh 
- src/www/httpd/htdocs/js/modules/camera_settings.js 
- src/www/httpd/htdocs/js/modules/ptz.js 
- src/www/httpd/htdocs/pages/camera_settings.html 
- src/www/httpd/htdocs/pages/ptz.html  
- src/www/httpd/htdocs/images/arrow-*.png 

Of course the src/www/httpd/htdocs/index.html has been modified to add the new link. I deleted the status page because there wasn't enough space on the nav menu.

I removed the changes I made to src/www/httpd/cgi-bin/get_configs.sh and src/www/httpd/cgi-bin/get_configs.sh. It was a copy of the files from the MStar repo, but these changes are not needed.

danielhainich commented:
Hi, iam not fimiliar using github. How I can Download your modified Firmware? I have yi Dome 1080p

snicker commented:
Just picked up a Yi-dome, would love to get PTZ built into this firmware. @Keroi do you have a release you have built already? Would save me the trouble of creating a build environment.

@TheCrypt0 do you have a timeline for integration of this feature to the mainline firmware?

iclems commented:
Very exciting to get control of the status led remotely.

odelma commented:
> Hi, iam not fimiliar using github. How I can Download your modified Firmware? I have yi Dome 1080p



> Just picked up a Yi-dome, would love to get PTZ built into this firmware. @Keroi do you have a release you have built already? Would save me the trouble of creating a build environment.

Check Keroi's forked repo here: https://github.com/Keroi/yi-hack-v4 (https://github.com/Keroi/yi-hack-v4/releases/tag/v0.2.1-h1). Hopefully someone with the needed skills would continue the work from there.
